ncm-ofed/src/main/perl/ofed.pm

```diff
@@ -13,31 +13,81 @@ use EDG::WP4::CCM::TextRender;
 use Readonly;
 
 Readonly my $OPENIB_CONF_TT => 'openib_conf';
+Readonly my $PARTITIONS_CONF_TT => 'partitions';
+Readonly my $NAMES_CONF_TT => 'names';
 
-sub Configure
-{
+Readonly my $PARTITIONS_FILENAME => '/etc/opensm/partitions.conf';
+Readonly my $NAMES_FILENAME => '/etc/opensm/ib-node-name-map';
 
-    my ($self, $config)=@_;
-
-    # write the openib config file (mandatory in schema)
-    my $cfg_fn = $config->getValue($self->prefix() . "/openib/config");
+sub _render
+{
+    my ($self, $element, $tt, $cfg_fn) = @_;
 
-    my $cfg_trd = EDG::WP4::CCM::TextRender->new($OPENIB_CONF_TT,
-                                             $config->getElement($self->prefix().'/openib'),
-                                             relpath => 'ofed',
-                                             log => $self);
+    my $cfg_trd = EDG::WP4::CCM::TextRender->new($tt,
+                                                 $element,
+                                                 relpath => 'ofed',
+                                                 log => $self);
 
     my $cfg_fh = $cfg_trd->filewriter($cfg_fn, log => $self, backup => ".old", mode => oct(644));
     if(! defined($cfg_fh)) {
         $self->error("Failed to render $cfg_fn: $cfg_trd->{fail}");
+        return;
     } else {
-        if($cfg_fh->close()) {
-            $self->verbose("ofed openib $cfg_fn modified, but openibd service not restarted (not yet supported by component)");
-            # TODO: support restart, but investigate impact first
-            # CAF::Service->new(['openibd'], log => $self)->restart();
-        }
+        return $cfg_fh->close();
+    }
+}
+
+sub openib
+{
+    my ($self, $config) = @_;
+
+    # write the openib config file (mandatory in schema)
+    my $cfg_fn = $config->getValue($self->prefix() . "/openib/config");
+    my $res = $self->_render($config->getElement($self->prefix().'/openib'),
+                             $OPENIB_CONF_TT,
+                             $cfg_fn);
+
+    if ($res) {
+        $self->verbose("ofed openib $cfg_fn modified, but openibd service not restarted (not yet supported by component)");
+        # TODO: support restart, but investigate impact first
+        # CAF::Service->new(['openibd'], log => $self)->restart();
+    }
+}
+
+sub opensm
+{
+    my ($self, $config) = @_;
+
+    my $tree = $config->getTree($self->prefix() . "/opensm");
+    my $changed = 0;
+
+    if ($tree->{partitions}) {
+        $changed += $self->_render($config->getElement($self->prefix().'/opensm/partitions'),
+                                   $PARTITIONS_CONF_TT,
+                                   $PARTITIONS_FILENAME)
+            || 0;
+    }
+
+    if ($tree->{names}) {
+        $changed += $self->_render($config->getElement($self->prefix().'/opensm/names'),
+                                   $NAMES_CONF_TT,
+                                   $NAMES_FILENAME)
+            || 0;
```

Comment:
i'll change it, but i don't care. someone who does should come up with perl tidy rules for stuff like this